webapp/javascript/components/Sidebar.tsx

```diff
@@ -110,7 +112,10 @@ export default function Sidebar2(props: SidebarProps) {
     </SubMenu>
   );
 
-  const toggleCollapse = () => setCollapsed(!collapsed);
+  const toggleCollapse = () => {
+    const action = collapsed ? uncollapseSidebar : collapseSidebar;
```

Comment:
Although this could've be written as a single action `toggleCollapse`, we prefer our actions to be idempotent, which makes it easier to understand.